### PR TITLE
refactor: make member functions const when applicable

### PR DESCRIPTION
**Pull request by aureleoules**
I have added a clang-tidy check: 'readability-make-member-function-const'.
> Finds non-static member functions that can be made const because the functions don’t use this in a non-const way.

I believe this makes the code more maintainable.

I used clang-tidy with `--fix-errors` and the check `readability-make-member-function-const` to generate the refactor commit.

### CODE REVIEW

**Review by jonatack — COMMENTED** [review]
Codebase-wide changes like this tend to be a difficult sell. Some initial thoughts.

Pro: 

- [By default make member functions const](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con2-by-default-make-member-functions-const)

Caveat:

- Const member functions need to be thread safe 

Related:

- [Don't make data members const or references](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c12-dont-make-data-members-const-or-references)

**Comment by MarcoFalke** [comment]
`std::move` will cast even mutable references, which makes hard-to-find bugs and is thus generally not wanted if the reference was passed to a function. So I don't think replacing forward with move is correct, unless you can provide a reason for the change.

**Comment by aureleoules** [comment]
> `std::move` will cast even mutable references, which makes hard-to-find bugs and is thus generally not wanted if the reference was passed to a function. So I don't think replacing forward with move is correct, unless you can provide a reason for the change.

my mistake, I've dropped the commit

**Comment by MarcoFalke** [comment]
Yeah, we should probably check this with https://clang.llvm.org/extra/clang-tidy/checks/bugprone/move-forwarding-reference.html

File: src/leveldb/util/env_posix.cc [hunk 1]
```diff
@@ -335,7 +335,7 @@ class PosixWritableFile final : public WritableFile {
     return status;
   }
 
-  Status WriteUnbuffered(const char* data, size_t size) {
+  Status WriteUnbuffered(const char* data, size_t size) const {
```
Comment (jonatack) [inline src/leveldb/util/env_posix.cc:338]: You are touching files in https://github.com/bitcoin-core/leveldb-subtree; changes to it need to be made there or upstream and not in bitcoin/bitcoin. See the CI linter failure: "FAIL: subtree directory was touched without subtree merge" (`test/lint/git-subtree-check.sh`)

Comment (aureleoules) [reply src/leveldb/util/env_posix.cc]: Thanks, I was wondering what this meant.
I removed these changes.

**Comment by fanquake** [comment]
I assume this is being done with some automated tooling? If so, could you expand the PR description to describe what you are doing to produce the change?

Similar to #25707, if you want to make a change like this, we'd likely also want this enforced/linted in some way. Are there any checks you can use in https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/list.html?

**Comment by jonatack** [comment]
Can tooling ensure these member functions marked as const are all thread-safe?

**Comment by MarcoFalke** [comment]
why the close? Seems a risk free CI check to avoid spending time on this during review?

**Comment by aureleoules** [comment]
@MarcoFalke I was asked during coredev to close this.

> This kind of code review-heavy codebase-wide refactor is not appropriate in general, especially for a new contributor. Constness can be accidental, and code/logical constness are not the same thing.

**Pull request closed without merging**